### PR TITLE
Remote config

### PR DESCRIPTION
**Pull request by suriksarkisyan**

### CODE REVIEW

File: android/src/main/kotlin/com/qonversion/flutter/sdk/qonversion_flutter_sdk/QonversionPlugin.kt [hunk 1]
```diff
@@ -263,6 +272,21 @@ class QonversionPlugin : MethodCallHandler, FlutterPlugin, ActivityAware {
         qonversionSandwich.checkTrialIntroEligibility(ids, result.toJsonResultListener())
     }
 
+    private fun attachUserToExperiment(args: Map<String, Any>, result: Result) {
+        @Suppress("UNCHECKED_CAST")
```
Comment (SpertsyanKM) [inline android/src/main/kotlin/com/qonversion/flutter/sdk/qonversion_flutter_sdk/QonversionPlugin.kt:276]: Does it really warn here (and below)? Seems like it shoudn't

File: example/lib/home.dart [hunk 2]
```diff
@@ -159,6 +159,8 @@ class _HomeViewState extends State<HomeView> {
         .build();
     Qonversion.initialize(config);
     Qonversion.getSharedInstance().collectAppleSearchAdsAttribution();
+    final dd = await Qonversion.getSharedInstance().remoteConfig();
+    print(dd);
```
Comment (SpertsyanKM) [inline example/lib/home.dart:163]:
```suggestion
```

File: lib/src/dto/remote_config.dart [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  final Map<String, dynamic> payload;
+
+  /// Object with the experiment's information.
+  @JsonKey(name: 'experiment')
+  final QExperiment? experiment;
+
+  const QRemoteConfig(
+      this.payload,
+      this.experiment,
+      );
```
Comment (SpertsyanKM) [inline lib/src/dto/remote_config.dart:20]:
```suggestion
  );
```

File: lib/src/qonversion.dart [hunk 4]
```diff
@@ -167,6 +167,12 @@ abstract class Qonversion {
   /// On iOS 14.0+ shows up a sheet for users to redeem AppStore offer codes.
   Future<void> presentCodeRedemptionSheet();
 
+  Future<QRemoteConfig> remoteConfig();
+
+  Future<void> attachUserToExperiment(String experimentId, String groupId);
+
+  Future<void> detachUserFromExperiment(String experimentId);
+
```
Comment (SpertsyanKM) [inline lib/src/qonversion.dart:175]: Comments

File: macos/Classes/SwiftQonversionPlugin.swift [hunk 5]
```diff
@@ -163,6 +172,10 @@ public class SwiftQonversionPlugin: NSObject, FlutterPlugin {
   private func checkEntitlements(_ result: @escaping FlutterResult) {
     qonversionSandwich?.checkEntitlements(getDefaultCompletion(result))
   }
+  
+  private func remoteConfig(_ result: @escaping FlutterResult) {
+    qonversionSandwich?.remoteConfig(getDefaultCompletion(result))
```
Comment (SpertsyanKM) [inline macos/Classes/SwiftQonversionPlugin.swift:177]: `getJsonCompletion`?

**Review by SpertsyanKM — APPROVED** [review]

File: android/src/main/kotlin/com/qonversion/flutter/sdk/qonversion_flutter_sdk/QonversionPlugin.kt (hunk 1 above)
Comment (suriksarkisyan) [reply android/src/main/kotlin/com/qonversion/flutter/sdk/qonversion_flutter_sdk/QonversionPlugin.kt:276]: It's a copy/paste from a similar function. I'll check and remove (if needed) it later.

**Pull request merged**